samples/wifi/provisioning/common/wifi_prov_ble.c

```diff
@@ -0,0 +1,416 @@
+);
+
+int wifi_prov_send_rsp(struct net_buf_simple *rsp)
+{
+	int rc;
+	static struct bt_gatt_indicate_params params;
+
+	params.attr = &prov_svc.attrs[PROV_SVC_CP_IDX];
+	params.data = rsp->data;
+	params.len = rsp->len;
+
+	if (!default_conn) {
+		LOG_INF("BT not connected. Ignore indication request.");
+		return 0;
+	}
+
+	if (bt_gatt_is_subscribed(default_conn, &prov_svc.attrs[PROV_SVC_CP_IDX],
+		BT_GATT_CCC_INDICATE)) {
+		rc = bt_gatt_indicate(default_conn, &params);
+		if (rc != 0) {
+			LOG_ERR("Indication error: %d.", rc);
+		}
+	} else {
+		LOG_INF("Wi-Fi Provisioning service - control point: cannot indicate");
+		return -EIO;
+	}
+	return rc;
+}
+
+int wifi_prov_send_result(struct net_buf_simple *result)
+{
+	if (!default_conn) {
+		LOG_INF("BT not connected. Ignore notification request.");
+		return 0;
+	}
+	if (bt_gatt_is_subscribed(default_conn, &prov_svc.attrs[PROV_SVC_DATA_IDX],
+		BT_GATT_CCC_NOTIFY)) {
+		return bt_gatt_notify(default_conn, &prov_svc.attrs[PROV_SVC_DATA_IDX],
+			result->data, result->len);
+	} else {
+		LOG_INF("Wi-Fi Provisioning service - data out: cannot notify");
+		return -EIO;
+	}
+}
+
+static void update_wifi_status_in_adv(void)
+{
+	int rc;
+	struct net_if *iface = net_if_get_default();
+	struct wifi_iface_status status = { 0 };
+
+	prov_svc_data[ADV_DATA_VERSION_IDX] = PROV_SVC_VER;
+
+	/* If no config, mark it as unprovisioned. */
+	if (!wifi_has_config()) {
+		prov_svc_data[ADV_DATA_FLAG_IDX] &= ~ADV_DATA_FLAG_PROV_STATUS_BIT;
+	} else {
+		prov_svc_data[ADV_DATA_FLAG_IDX] |= ADV_DATA_FLAG_PROV_STATUS_BIT;
+	}
+
+	rc = net_mgmt(NET_REQUEST_WIFI_IFACE_STATUS, iface, &status,
+				sizeof(struct wifi_iface_status));
+	/* If WiFi is not connected or error occurs, mark it as not connected. */
+	if ((rc != 0) || (status.state < WIFI_STATE_ASSOCIATED)) {
+		prov_svc_data[ADV_DATA_FLAG_IDX] &= ~ADV_DATA_FLAG_CONN_STATUS_BIT;
+		prov_svc_data[ADV_DATA_RSSI_IDX] = INT8_MIN;
+	} else {
+		/* WiFi is connected. */
+		prov_svc_data[ADV_DATA_FLAG_IDX] |= ADV_DATA_FLAG_CONN_STATUS_BIT;
+		/* Currently cannot retrieve RSSI. Use a dummy number. */
+		prov_svc_data[ADV_DATA_RSSI_IDX] = status.rssi;
+	}
+
+	rc = bt_le_adv_update_data(ad, ARRAY_SIZE(ad), sd, ARRAY_SIZE(sd));
+	if (rc != 0) {
+		LOG_DBG("Cannot update advertisement data, err = %d", rc);
+	}
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	if (err) {
+		LOG_WRN("BT Connection failed (err 0x%02x).", err);
+		return;
+	}
+
+	default_conn = conn;
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	LOG_INF("BT Connected: %s", addr);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	LOG_INF("BT Disconnected: %s (reason 0x%02x).", addr, reason);
+	bt_unpair(BT_ID_DEFAULT, NULL);
+	default_conn = NULL;
+	k_work_reschedule(&update_adv_param_work, K_NO_WAIT);
```

Comment:
What about `update_adv_data_work`? Should it be restarted here?

Reply:
Added. Now the task for updating adv data will stop when connected, and start when disconnected.